dbt_platform_helper/utils/validation.py

```diff
@@ -274,6 +274,29 @@ def is_between(value):
     "enabled": bool,
 }
 
+SOURCE_KMS_KEY_ARN = Regex(
+    r"^arn:aws:kms:.*:\d{12}:(key|alias).*",
+    error="source_kms_key_arn must be a valid arn for a kms key",
+)
+
+SOURCE_BUCKET_ARN = Regex(
+    r"^arn:aws:s3::.*", error="source_bucket_arn must be a valid arn for an s3 bucket"
+)
+
+WORKER_ROLE_ARN = Regex(
+    r"^arn:aws:iam::\d{12}:role/.*", error="worker_role_arn must be a valid arn for an iam role"
+)
+
+DATA_IMPORT = {
+    Optional("source_kms_key_arn"): SOURCE_KMS_KEY_ARN,
+    "source_bucket_arn": SOURCE_BUCKET_ARN,
+    "worker_role_arn": WORKER_ROLE_ARN,
+}
```

Comment:
unfortunately not, which is why I ended up using the key name in the error message... I can take another look if there's a generic way to do it but still get an informative enough error

Reply:
I've pushed another suggestion which would allow passing in the key so you can see which property triggered the error.  Still not really happy with the pattern, but I feel like it could turn into a bigger job to try and build a nice validator class.  What do you think?